CIP-0035/README.md

```diff
@@ -240,10 +261,10 @@ However, this becomes less so as the platform starts to mature and is neither su
 Furthermore, while many changes to Cardano are obscure or not of interest to many community members, there is a much larger community who have a keen interest in changes to Plutus Core: dApp developers. 
 Hence it is especially important to have a clear way for this community to be able to propose changes and see how they are progressing.
 
-### Do removals and changes really need a new language version?
+### Do removals and changes really need a new ledger language?
 
-Not being able to make removals or changes to behaviour without a LV is quite painful. 
-For example, it means that we cannot just fix bugs in the semantics: we must remain bug-for-bug compatible with any given LV.
+Not being able to make removals or changes to behaviour without a LL is quite painful. 
+For example, it means that we cannot just fix bugs in the semantics: we must remain bug-for-bug compatible with any given LL.
```

Comment:
What do we do if we discover that some builtin contains a bug that can compromise the chain, like a division by zero error in a C library?

Also, what do we do if there's a change in the behaviour of the (Haskell or C) library function underlying a builtin at some point in the future? Would we have to maintain a version of the library function reproducing the old behaviour in perpetuity and use it for the execution of existing scripts depending on the old behaviour?

Reply:
I think that's all out of scope of this CIP (since I would say that's "emergency measures"), but
1. No script can ever depend on behaviour that crashes the node. So we can deal with such things. If there's wrong but non-crashing behaviour... we need to preserve that.
2. Yes, if there's a change in the behaviour of the underlying function we'll need to maintain a version with the old behaviour. I guess that's another criteria for builtins: the behaviour should be as clearly specified as possible, so it won't change.